src/blitter/40bpp_anim.cpp

```diff
@@ -240,26 +241,42 @@ inline void Blitter_40bppAnim::Draw(const Blitter::BlitterParams *bp, ZoomLevel
 
 					/* Make the current colour a bit more black, so it looks like this image is transparent */
 					src_n += n;
-					if (src_px->a == 255) {
-						src_px += n;
-						do {
-							/* If the anim buffer contains a color value, the image composition will
-							 * only look at the RGB brightness value. As such, we can simply darken the
-							 * RGB value to darken the anim color. */
-							Colour b = *anim != 0 ? Colour(this->GetColourBrightness(*dst), 0, 0) : *dst;
-							*dst = this->MakeTransparent(b, 3, 4);
-							anim++;
-							dst++;
-						} while (--n != 0);
+					if (bp->transparent) {
+						if (src_px->a == 255) {
+							src_px += n;
+							do {
+								/* If the anim buffer contains a color value, the image composition will
+								* only look at the RGB brightness value. As such, we can simply darken the
+								* RGB value to darken the anim color. */
```

Comment:
Your editor removed the space before the `*` in the last 2 lines :D

I know this issue all too well, and I absolutely hate my editor does that :P